src/phylo/phylogeny.jl

```diff
@@ -0,0 +1,230 @@
+(Float64) for the clades.
+
+# Examples
+```julia
+Phylogeny([:Human, :Chimp, :Dog, :Cat])
+```
+"""
+function Phylogeny(taxa::Vector{Symbol})
+    return Phylogeny(taxa, Float64, Float64)
+end
+
+
+"""
+    Phylogeny{T <: AbstractString}(taxa::Vector{T})
+Construct a completely unresolved phylogeny, with no annotated data.
+
+This constructor assumes you want a phylogeny where the only data associated
+with it is branch lengths (Float64) for the edges, and confidence values
+(Float64) for the clades.
+"""
+function Phylogeny{T <: AbstractString}(taxa::Vector{T})
+    return Phylogeny(convert(Vector{Symbol}, taxa))
+end
+
+
+function Base.copy{C, B}(tree::Phylogeny{C, B})
+    g = copy(tree.graph)
+    i = copy(tree.vertexindex)
+    v = copy(tree.vertexdata)
+    d = copy(tree.edgedata)
+    return Phylogeny{C, B}(g, i, v, d, tree.ntaxa, tree.rooted, tree.rerootable)
+end
+
+
+# Miscellaneous Functions
+# ------------------------
+
+function n_possible_rooted(n::Int)
+    @assert n >= 2 "n must be greater or equal to 2."
+    num = factorial(2 * n - 3)
+    den = (2^(n - 2)) * factorial(n - 2)
+    return Int(num / den)
+end
+
+function n_possible_unrooted(n::Int)
+    @assert n >= 3 "n must be greater of equal to 3."
+    return n_possible_rooted(n - 1)
+end
+
+max_vertices(n::Int) = 2 * n - 1
+
+
+typealias SymString Union{Symbol, AbstractString}
+
+
+"""
+    is_rooted(x::Phylogeny)
+
+Test whether a Phylogeny is rooted.
+
+# Examples
+```julia
+is_rooted(my_phylogeny)
+```
+"""
+function is_rooted(x::Phylogeny)
+    return x.rooted
+end
+
+
+"""
+    is_rerootable(x::Phylogeny)
+
+Test whether a Phylogeny is re-rootable.
+
+# Examples
+```julia
+is_rerootable(my_phylogeny)
+```
+"""
+function is_rerootable(x::Phylogeny)
+    return x.rerootable
+end
+
+
+"Get the vertex of the tree which represents the root of the tree."
+root_vertex(tree::Phylogeny) = tree.ntaxa + 1
+
+
+"Get the range of vertices which represent the internal clades of the tree."
+clade_vertices(tree::Phylogeny) = (root_vertex(tree) + 1):nv(tree.graph)
+
+
+"Get the range of vertices which represent the leaves of the tree."
+leaf_vertices(tree::Phylogeny) = 1:tree.ntaxa
+
+
+
+"""
+    has_parent(tree::Phylogeny)
+
+Check every vertex in a phylogeny and test if each is linked to a parent.
+"""
+function has_parent(tree::Phylogeny)
+    return indegree(tree.graph) .== 1
+end
+
+
+"""
+    no_parent(tree::Phylogeny)
+
+Check every vertex in a phylogeny and test if each is not linked to a parent.
+"""
+function no_parent(tree::Phylogeny)
+    return indegree(tree.graph) .== 0
+end
+
+
+"""
+    has_children(tree::Phylogeny)
+
+Test each vertex in a phylogeny to see if it has children.
+"""
+function has_children(tree::Phylogeny)
+    return n_children(tree) .> 0
+end
+
+
+"""
+    no_children(tree::Phylogeny)
+
+Test each vertices in a phylogeny to see if it lacks children.
+"""
+function no_children(tree::Phylogeny)
+    return n_children(tree) .== 0
+end
```

Comment:
`has_parent`, `no_parent`, `has_children`, and `no_children` for the `Phylogeny` type are not tested.